cloudant-client/src/test/java/com/cloudant/api/query/IndexListTests.java

```diff
@@ -0,0 +1,249 @@
+/*
+ * Copyright © 2017 IBM Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+package com.cloudant.api.query;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+import com.cloudant.client.api.query.Field;
+import com.cloudant.client.api.query.Index;
+import com.cloudant.client.api.query.JsonIndex;
+import com.cloudant.client.api.query.Sort;
+import com.cloudant.client.api.query.TextIndex;
+import com.cloudant.tests.util.MockedServerTest;
+
+import org.junit.Test;
+
+import okhttp3.mockwebserver.MockResponse;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class IndexListTests extends MockedServerTest {
+
+    private static String SELECTOR_STRING = "{\"year\":{\"$gt\":2010}}";
+
+    // _all_docs
+    private static String ALL_DOCS = "{\"ddoc\":null,\"name\":\"_all_docs\",\"type\":\"special\"," +
+            "\"def\":{\"fields\":[{\"_id\":\"asc\"}]}}";
+    private static String JSON_SIMPLE = "{\"ddoc\":\"_design/testindexddoc\"," +
+            "\"name\":\"simplejson\",\"type\":\"json\"," +
+            "\"def\":{\"fields\":[{\"Person_dob\":\"asc\"}],\"partial_filter_selector\":{}}}";
```

Comment:
Since we're testing deserialisation, there ought to be one example which uses `selector` instead of `partial_filter_selector`

Reply:
Added in b36b7b4